### DISCUSSION TITLE
CSLA 6 prerelease 7 online

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#2836] [Announcements]
I am pleased to let everyone know that CSLA 6 prerelease 7 is now in NuGet.

Thanks to @swegele and @TheCakeMonster, this release fixes a complex challenge with how `LocalProxy` creates a new DI scope, and how context flows to that scope from the logical client. The testing by @swegele and insights from both @swegele and @TheCakeMonster were invaluable in this effort!

`PropertyChangedMode` now defaults to `Windows` for Blazor apps, a change made thanks to some great community feedback and testing. It turns out that Blazor data binding can be very chatty (causing perf issues) if `PropertyChanged` is raised for every property that's changed. Fortunately, Windows Forms has similar issues, and so CSLA already has this mode where `PropertyChanged` events are minimized, while still ensuring that at least _one_ gets raised to trigger a UI refresh.

The business and authorization rule context objects now properly provide access to the data portal for use in your rules.

There is now an `IDataPortalFactory` service that you can inject, allowing you to easily create multiple `IDataPortal<T>` instances as needed.

You can now (optionally) get a `IAddObjectAuthorizationRulesContext` parameter in the static `AddObjectAuthorizationRulesContext` method, so you can use context and the data portal to load your per-type authz rules.

There may be other changes as well - these are the highlights. Every one of which was brought up by community members who've been testing and providing feedback:

**Thank you all!!**

**Comment by ashishsinha24** [comment]
There was a Prelease 6 a few days back. Shouldn't it be Prerelease 7?

**Reply by rockfordlhotka** [reply]
😳

**Reply by TheCakeMonster** [reply]
What's a digit between friends?

6.1?
6i?
6(better)?

😉

**Reply by TheCakeMonster** [reply]
Six and a bit. Yeah, that works.
